core/src/main/java/com/google/zxing/aztec/encoder/HighLevelEncoder.java

```diff
@@ -148,16 +151,31 @@
   }
 
   private final byte[] text;
+  private final Charset charset;
 
   public HighLevelEncoder(byte[] text) {
     this.text = text;
+    this.charset = null;
+  }
+
+  public HighLevelEncoder(byte[] text, Charset charset) {
+    this.text = text;
+    this.charset = charset;
   }
 
   /**
    * @return text represented by this encoder encoded as a {@link BitArray}
    */
   public BitArray encode() {
-    Collection<State> states = Collections.singletonList(State.INITIAL_STATE);
+    State initialState = State.INITIAL_STATE;
+    if (null != charset) {
```

Comment:
Nit: I usually write `foo != null` in this code base
Also 2-space indent rather than 4 below